Src/BootCamp.Chapter/Lesson3.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Globalization;
+
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+         public static string ReadInputString(string message)
+        {
+            Console.WriteLine(message);
+            string value = Console.ReadLine();
+            
+            if (string.IsNullOrEmpty(value))
+            {
+                Console.Write("Name cannot be empty.");
+                value = "-";
+            }
+
+            return value;
+        }
+
+        public static float ReadInputFloat(string message)
+        {
+            Console.WriteLine(message);
+            string input = Console.ReadLine();
+            float value = ValidateFloat(input);
+            if (value < 0) PrintInvalidNumber(input);
+
+            return value;
+        }
+
+        public static int ReadInputInt(string message)
+        {
+            Console.WriteLine(message);
+            string input = Console.ReadLine();
+            int value = ValidateInt(input);
+            if (value < 0) PrintInvalidNumber(input);
+            
+            return value; 
+        }
+
+        static int ValidateInt(string input)
+        {
+            int result;
+            if (string.IsNullOrEmpty(input))
+                result = 0;
+            else
+                if (!int.TryParse(input, out result))
+                    result = -1;
+            return result;
+        }
+
+        static float ValidateFloat(string input)
+        {
+            float result;
+            if (string.IsNullOrEmpty(input))
+                result = 0;
+            else
+                if (!float.TryParse(input, NumberStyles.Float, CultureInfo.InvariantCulture, out result))
+                result = -1;
+            return result;
+        }
+
+        static void PrintInvalidNumber(string input)
+        {
+            Console.Write("\"" + input + "\" is not a valid number.");
+        }
+
+        public static float CalculateBmi(float height, float weight)
+        {
+            if (height <= 0 || weight <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+                if (weight <= 0)
+                    Console.WriteLine("Weight cannot be equal or less than zero, but was " + weight + ".");
+                if (height == 0 && weight == 0)  
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917496)